### PR TITLE
fix for mac m1 && add some mix

### PR DESCRIPTION
**Pull request by piz-ewing**
now, it works fine

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 .idea/
 target
-
+Cargo.lock
```
Comment (bjorn3) [inline .gitignore]: Why did you add `Cargo.lock` to `.gitignore`?

File: Cargo.toml [hunk 2]
```diff
@@ -5,10 +5,10 @@ authors = ["Xargin <cao1988228@163.com>"]
 edition = "2018"
 
 [dependencies]
-capstone = "^0.6.0"
-keystone = "0.9.0"
-unicorn = "0.9.1"
+capstone = "^0.9.0"
+keystone = {git = "https://github.com/keystone-engine/keystone/"}
+unicorn-engine = {git = "https://github.com/unicorn-engine/unicorn/", branch = "dev"}
```
Comment (bjorn3) [inline Cargo.toml]: Maybe specify a specific commit for those git dependencies?

File: .gitignore (hunk 1 above)
Comment (piz-ewing) [reply .gitignore]: it's useless，I will remove it

File: Cargo.toml (hunk 2 above)
Comment (piz-ewing) [reply Cargo.toml]: I want to keep them up to date until they don't work

Comment (piz-ewing) [reply Cargo.toml]: maybe i should specify a specific commit id

**Comment by piz-ewing** [comment]
#2 Solve the missing static library

another way for Mac m1:
1. modify Cargo.toml for fix bug
```
[patch.crates-io]
xxxx
```
**keystone bug**
Fix use after free in Rust bindings (https://github.com/keystone-engine/keystone/commit/28292239523d5bcdcac267775c9882c84d5cfec1)

**change BINDINGS_MAJOR = 2**
[libunicorn-sys](https://github.com/unicorn-rs/unicorn-rs/blob/125adc1cd7bfdcab9bfd1e37c8eb1456fea825eb/libunicorn-sys/src/unicorn_const.rs#L181)

2. use brew install unicorn2

[unicorn2.rb.tar.gz](https://github.com/cch123/asm-cli-rust/files/8563519/unicorn2.rb.tar.gz)
you should create `/opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/unicorn2.rb`

3. add config.toml
```
[build]
rustflags = " -L/opt/homebrew/Cellar/unicorn/2.0.0/lib -L/opt/homebrew/Cellar/keystone/0.9.2/lib/"
```

**Comment by cch123** [comment]
Thx very much!

**Pull request merged**